### PR TITLE
Proposal for integration with Laravel casts

### PR DESCRIPTION
**Pull request by khepin**
This shows how we could provide an integration for casting to and from
json with actual PHP objects for Laravel's Eloquent models.

Looking for feedback.

@mattgrande @bezhermoso @szainmehdi

### CODE REVIEW

File: src/Integrations/Laravel/JsonCastable.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php declare(strict_types=1);
+
+namespace Square\Pjson\Integrations\Laravel;
+
+trait JsonCastable
+{
+    public static function castUsing(array $arguments)
+    {
+        return new JsonCaster(static::class);
+    }
```
Comment (bezhermoso) [inline src/Integrations/Laravel/JsonCastable.php:10]: 😗 🤌

**Review by bezhermoso — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -481,3 +481,70 @@ that this is expected behavior by adding this library's extension in your `phpst
...
+    'schedule' => Schedule::class,
+];
+```
+
+### via cast arguments
+
+Alternatively, you can simply use Laravel's cast arguments. In this case the `Schedule` class stays the way it used to be:
+
+```php
+class Schedule implements
```
Comment (bezhermoso) [inline README.md:526]:
```suggestion
class Schedule
```

File: README.md [hunk 3]
```diff
@@ -481,3 +481,70 @@ that this is expected behavior by adding this library's extension in your `phpst
...
+
+If you wish to cast Eloquent model attributes to classes via Pjson, you might do so with the provided casting utilities:
+
+```php
+use Illuminate\Contracts\Database\Eloquent\Castable;
+
+class Schedule implements Castable // implement the laravel interface
+{
+    use JsonSerialize;
+    use JsonCastable; // use the provided Pjson trait
```
Comment (mattgrande) [inline README.md:497]: I think it's worthwhile to include the `use` statements for these traits as well.

**Review by mattgrande — APPROVED** [review]
Love love love this! Added a nitpick to the readme, but otherwise, it looks good!

File: tests/Integrations/Laravel/LaravelCastTest.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        });
+    }
+
+    public function testCasts()
+    {
+        $addr = '{"line1":"678 Lombard St.","line2":"","city":"San Fransokyo","zipcode":"94959","country":"USA"}';
+        (new FirstModel([
+            'name' => 'jane',
+            'castable_address' => CastableAddress::fromJsonString($addr),
+            'address' => Address::fromJsonString($addr),
```
Comment (bezhermoso) [inline tests/Integrations/Laravel/LaravelCastTest.php:29]: @khepin Can we add coverage for null handling?

**Pull request merged**